group-coordinator/src/main/java/org/apache/kafka/coordinator/group/streams/StreamsGroupMember.java

```diff
@@ -0,0 +1,756 @@
+            setTopologyEpoch(record.topologyEpoch());
+            setProcessId(record.processId());
+            setUserEndpoint(record.userEndpoint());
+            setClientTags(record.clientTags().stream().collect(Collectors.toMap(
+                StreamsGroupMemberMetadataValue.KeyValue::key,
+                StreamsGroupMemberMetadataValue.KeyValue::value
+            )));
+            return this;
+        }
+
+        public Builder updateWith(StreamsGroupCurrentMemberAssignmentValue record) {
+            setMemberEpoch(record.memberEpoch());
+            setPreviousMemberEpoch(record.previousMemberEpoch());
+            setState(MemberState.fromValue(record.state()));
+            setAssignedActiveTasks(assignmentFromTaskIds(record.activeTasks()));
+            setAssignedStandbyTasks(assignmentFromTaskIds(record.standbyTasks()));
+            setAssignedWarmupTasks(assignmentFromTaskIds(record.warmupTasks()));
+            setActiveTasksPendingRevocation(
+                assignmentFromTaskIds(record.activeTasksPendingRevocation()));
+            setStandbyTasksPendingRevocation(
+                assignmentFromTaskIds(record.standbyTasksPendingRevocation()));
+            setWarmupTasksPendingRevocation(
+                assignmentFromTaskIds(record.warmupTasksPendingRevocation()));
+            return this;
+        }
+
+        private Map<String, Set<Integer>> assignmentFromTaskIds(
+            List<StreamsGroupCurrentMemberAssignmentValue.TaskIds> topicPartitionsList
+        ) {
+            return topicPartitionsList.stream().collect(Collectors.toMap(
+                StreamsGroupCurrentMemberAssignmentValue.TaskIds::subtopologyId,
+                taskIds -> Collections.unmodifiableSet(new HashSet<>(taskIds.partitions()))));
+        }
+
+        public StreamsGroupMember build() {
+            return new StreamsGroupMember(
+                memberId,
+                memberEpoch,
+                previousMemberEpoch,
+                instanceId,
+                rackId,
+                rebalanceTimeoutMs,
+                clientId,
+                clientHost,
+                topologyEpoch,
+                processId,
+                userEndpoint,
+                clientTags,
+                state,
+                assignedActiveTasks,
+                assignedStandbyTasks,
+                assignedWarmupTasks,
+                activeTasksPendingRevocation,
+                standbyTasksPendingRevocation,
+                warmupTasksPendingRevocation
+            );
+        }
+    }
+
+    /**
+     * The member id.
+     */
+    private final String memberId;
+
+    /**
+     * The current member epoch.
+     */
+    private final int memberEpoch;
+
+    /**
+     * The previous member epoch.
+     */
+    private final int previousMemberEpoch;
+
+    /**
+     * The member state.
+     */
+    private final MemberState state;
+
+    /**
+     * The instance id provided by the member.
```

Comment:
probably use `ID` instead of `id` everywhere in the comments.

Reply:
Done